src/tools/FirmwareUpdater/mainwindow.cpp

```diff
@@ -1001,6 +1008,8 @@ void MainWindow::onAppendInfo(boardInfo2_t info,eOipv4addr_t address)
     infoTreeWidget->addTopLevelItem(bootStrapNode);
     infoTreeWidget->addTopLevelItem(propertiesNode);
 
+    constexpr uint8_t maxNumberOfProcessesOnSingleCore {3};

```

Comment:
I can see that later on we have a switch(info.boardtype) where it is missing the case of amc. 
Some comments:
- the string name in here is wrongly obtained by a switch case and we don't have `amc`. There is another mode of obtaining the string and is by using the relevant function inside icub-firmware-shared. We should stick to this latter otherwise we miss the support for new boards.
- At the same mode, if we want to know if the board has two cores, it is wise to add a function inside icub-firmware-shared.

Reply:
fixed (partially?) in https://github.com/robotology/icub-main/pull/850/commits/f0b7ded1289f8b95e25d41e18eaea21a61070232